analyzer/evmabibackfill/evm_abi_backfill_test.go

```diff
@@ -0,0 +1,298 @@
+
+	expected := []mockParsedEvent{
+		{
+			Name: &name1,
+			Sig:  &sig1,
+			Args: []*abiEncodedArg{
+				{
+					Name:    "src",
+					EvmType: "address",
+					Value:   ethCommon.HexToAddress("0xc37F8341Ac6e4a94538302bCd4d49Cf0852D30C0"),
+				},
+				{
+					Name:    "dst",
+					EvmType: "address",
+					Value:   ethCommon.HexToAddress("0x59989A8fBff8be3d1deAFFF2b46B66B63ACB797e"),
+				},
+				{
+					Name:    "wad",
+					EvmType: "uint256",
+					Value:   "876558921078386560",
+				},
+			},
+		},
+		{
+			Name: &name2,
+			Sig:  &sig2,
+			Args: []*abiEncodedArg{
+				{
+					Name:    "src",
+					EvmType: "address",
+					Value:   ethCommon.HexToAddress("0x5C72F76F7dae16dD34Cb6183b73F4791aa4B3BC4"),
+				},
+				{
+					Name:    "wad",
+					EvmType: "uint256",
+					Value:   "14212523248195733504",
+				},
+			},
+		},
+		{
+			Name: &name3,
+			Sig:  &sig3,
+			Args: []*abiEncodedArg{
+				{
+					Name:    "dst",
+					EvmType: "address",
+					Value:   ethCommon.HexToAddress("0xDd1Ee07b46C7eD888671a646F3c7a37394e8cF85"),
+				},
+				{
+					Name:    "wad",
+					EvmType: "uint256",
+					Value:   "10000000000000000000000",
+				},
+			},
+		},
+		{
+			Name: &name4,
+			Sig:  &sig4,
+			Args: []*abiEncodedArg{
+				{
+					Name:    "src",
+					EvmType: "address",
+					Value:   ethCommon.HexToAddress("0xDd1Ee07b46C7eD888671a646F3c7a37394e8cF85"),
+				},
+				{
+					Name:    "guy",
+					EvmType: "address",
+					Value:   ethCommon.HexToAddress("0xb539f1D01A437C7f30cAfC994e918F952dDc0bA2"),
+				},
+				{
+					Name:    "wad",
+					EvmType: "uint256",
+					Value:   "10000000000000000000000",
+				},
+			},
+		},
+	}
+
+	for i, ev := range evs {
+		name, args, sig, err := p.parseEvent(ev, *abi)
+		require.Nil(t, err)
+		verifyEvent(t, &expected[i], name, args, sig)
+	}
+}
+
+func TestParseUnknownEvent(t *testing.T) {
+	abi := evmabi.WROSE
+	p := &processor{
+		runtime: common.RuntimeSapphire,
+		target:  nil,
+		logger:  log.NewDefaultLogger("testing"),
+	}
+	// Non-WROSE event.
+	rawEv := []byte(`{"data": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=", "topics": ["8nnmofXjIMypETVnbZy25EyooIwLiDQrzbEUT2URtWg=", "AAAAAAAAAAAAAAAArk2RKpUohmFUB3UShFG6FIKuqKc=", "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAI="], "address": "jZzJ7hGq+GWRPe7JOe6y3Hg4q3s="}`)
+	ev := unmarshalEvmEvent(t, rawEv)
+
+	name, args, sig, err := p.parseEvent(ev, *abi)
+	require.NotNil(t, err)
+	require.Nil(t, name)
+	require.Nil(t, args)
+	require.Nil(t, sig)
+}
+
+func TestParseTransactionCall(t *testing.T) {
+	abi := evmabi.WROSE
+	p := &processor{
+		runtime: common.RuntimeSapphire,
+		target:  nil,
+		logger:  log.NewDefaultLogger("testing"),
+	}
+	txs := []*abiEncodedTx{
+		// unmarshalEvmTx(t, "omRib2R5o2Jwa1gg6KR1JehYdjBZyYOH06CycJIxAuTmmY22C6nLtybyPTlkZGF0YVgbjk8cKK1D/PuiREnFTOkHm4YosIhcDHtRzPkPZW5vbmNlT95YOkkjR3eTTH7QR/O8tmZmb3JtYXQB", nil), // The WROSE abi in the nexus repository is out of date/incomplete.
```

Comment:
What does this base64 represent? Unless it covers an important new aspect of ABI parsing, let's just remove it from here.

Reply:
It's a more complex tx call from the actual wrose abi; but it shouldn't test any new codepaths so I'll remove it. Can revisit if/when we update the wrose abi in nexus.